lib/spack/spack/solver/asp.py

```diff
@@ -3134,20 +3134,6 @@ def setup(
         self.pkgs = node_counter.possible_dependencies()
         self.libcs = sorted(all_libcs())  # type: ignore[type-var]
 
-        # Fail if we already know an unreachable node is requested
-        for spec in specs:
-            # concrete roots don't need their dependencies verified
-            if spec.concrete:
-                continue
-
-            missing_deps = [
-                str(d)
-                for d in spec.traverse()
-                if d.name not in self.pkgs and not spack.repo.PATH.is_virtual(d.name)
-            ]
-            if missing_deps:
-                raise spack.spec.InvalidDependencyError(spec.name, missing_deps)
```

Comment:
This was failing on:
```
foo %[when=%mpi virtuals=mpi] mpich
```
if `foo` couldn't depend on `mpi`. Since the check has been rarely useful, I removed it instead of making it more complicated by iterating on edges.